pkg/controllers/manifestwork.go

```diff
@@ -39,11 +39,17 @@ const (
 	NamespaceNameSeperator        = "/"
 )
 
-func ScafoldManifestwork(hyd *hypdeployment.HypershiftDeployment) *workv1.ManifestWork {
+func ScaffoldManifestwork(hyd *hypdeployment.HypershiftDeployment) (*workv1.ManifestWork, error) {
+	if len(hyd.Spec.InfraID) == 0 {
+		return nil, fmt.Errorf("hypershiftDeployment.Spec.InfraID is not set or rendered")
```

Comment:
Do we want to consider logging the error and exiting with `nil` so reconcile stops till someone makes a change?